src/controllers/resource.ts

```diff
@@ -139,4 +157,114 @@ export class ResourceController {
 			});
 		}
 	}
+	public async createResource(request: Request, response: Response) {
+		const result = validationResult(request);
+		if (!result.isEmpty()) {
+			return response
+				.status(400)
+				.json(Responses.GetInvalidResourceResponse('', {}, request.body.secret, result.array()[0].msg));
+		}
+
+		let {
+			did,
+			jobId,
+			content,
+			name,
+			type,
+			version,
+			secret = {},
+			options = {},
+		} = request.body as IResourceCreateRequest;
+
+		let resourcePayload: Partial<MsgCreateResourcePayload> = {};
+		try {
+			// check if did is registered on the ledger
+			let resolvedDocument = await CheqdResolver(did);
+			if (!resolvedDocument?.didDocument || resolvedDocument.didDocumentMetadata.deactivated) {
+				return response
+					.status(400)
+					.send(Responses.GetInvalidResponse({ id: did }, secret, Messages.DidNotFound));
+			} else {
+				resolvedDocument = resolvedDocument.didDocument;
+			}
+
```

Comment:
To discuss: should we add an additional check to if the resource name and type already exist. That would prevent someone from updating a resource using the create endpoint.

Reply:
Lets block update-resource via createResource.